packages/format-library/src/link/index.js

```diff
@@ -39,6 +40,42 @@ function Edit( {
 	contentRef,
 } ) {
 	const [ addingLink, setAddingLink ] = useState( false );
+	const [ clickedLink, setClickedLink ] = useState( false );
+
+	const anchorElement = useAnchor( {
+		editableContentElement: contentRef.current,
+		settings: link,
+	} );
+
+	function setClickedLinkTrue() {
+		setClickedLink( true );
+	}
+
+	function setClickedLinkFalse() {
+		setClickedLink( false );
+	}
+
+	useLayoutEffect( () => {
+		// log tagNAME of anchorElement
+		if ( anchorElement?.tagName?.toLowerCase() === 'a' ) {
```

Comment:
Also if anchorElement is defined and isActive is true, it's guaranteed to be a link no?

Reply:
No because it can be a VirtualAnchorElement returned from `useAnchor`.

I've ended up avoiding the `useAnchor` entirely now by attaching listeners to the contenteditable.

Reply:
> No because it can be a VirtualAnchorElement returned from useAnchor.

How can I test VirtualAnchorElements? What blocks use them?

Reply:
You can't 😄 Take a look at `useAnchor` and you'll see it returns a "virtual" anchor representing a selection within RichText. So for example, when you highlight some text and you want to make a link, the Link UI needs to have an anchor point to "attach" the Popover to, but the link (`<a>` TAG) doesn't yet exist in the DOM. So `useAnchor` will return a coordinate which approximates the selection in the richtext.

Reply:
Nice work!

We'll need some new e2e test coverage for these changes. Specifically it would be super easy for regressions to be introduced around this new behaviour so we need to validate the new expectations.

Reply:
I think fixing all the broken e2e tests will validate it too.